Unable to build Cyclone DDS on Ubuntu 18.04

I try to build Cyclone DDS on Ubuntu 18.04.
However, I find that there is the error message while building.

```bash
$ git clone https://github.com/eclipse-cyclonedds/cyclonedds.git
$ cd cyclonedds
$ mkdir build && cd build
$ cmake ..

......(ignore some message)

-- Found Maven: /usr/bin/mvn (found suitable version "3.6.0", minimum required is "3.0") 
-- Found Java: /usr/bin/java (found suitable version "11.0.7", minimum required is "1.8") 
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
CMake Error at /usr/share/cmake-3.10/Modules/CheckCXXSourceCompiles.cmake:97 (try_compile):
  Unknown extension ".cxx" for file

    /home/chenying/workspace/cyclonedds/build/CMakeFiles/CMakeTmp/src.cxx

  try_compile() works only for enabled languages.  Currently these are:

    C

  See project() command to enable other languages.
Call Stack (most recent call first):
  /usr/share/cmake-3.10/Modules/CheckCXXCompilerFlag.cmake:49 (CHECK_CXX_SOURCE_COMPILES)
  /usr/share/cmake-3.10/Modules/GenerateExportHeader.cmake:215 (check_cxx_compiler_flag)
  /usr/share/cmake-3.10/Modules/GenerateExportHeader.cmake:374 (_test_compiler_hidden_visibility)
  src/security/builtin_plugins/access_control/CMakeLists.txt:23 (generate_export_header)


-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Failed
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY

......(ignore some message)
```

I guess the error is related to [the issue](https://gitlab.kitware.com/cmake/cmake/-/issues/16967) in CMAKE project. It's fixed in [the PR](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/1799/diffs).
Maybe CMAKE in Ubuntu 18.04 haven't had the bugfix yet.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,6 +11,10 @@
 #
 cmake_minimum_required(VERSION 3.7)
 project(CycloneDDS VERSION 0.7.0 LANGUAGES C)
+if(CMAKE_VERSION VERSION_LESS 3.12)
+  # GENERATE_EXPORT_HEADER requires a C++ compiler up to version 3.12
+  enable_language(CXX)
+endif()
 
 # Set a default build type if none was specified
 set(default_build_type "RelWithDebInfo")

